File: app/dbmodels/Software.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Software',
    'menu'          => 'Asset'
);
$table['fields'] = array(
    'id'                   => DBModels::type('increments',
                                             array('visible' => false)),
    'name'                 => DBModels::type('string'),
    'comment'              => DBModels::type('text'),
    'is_deleted'           => DBModels::type('boolean',
                                             array('visible' => false)),
    'is_template'          => DBModels::type('boolean',
                                             array('visible' => false)),
    'template_name'        => DBModels::type('string',
                                             array('visible' => false)),
    'user_id'              => DBModels::type('integer'),
    'user_tech_id'         => DBModels::type('integer'),
    'group_id'             => DBModels::type('integer'),
    'group_tech_id'        => DBModels::type('integer'),
    'location_id'          => DBModels::type('integer'),
    'software_id'          => DBModels::type('integer'),
    'softwarecategory_id'  => DBModels::type('integer'),
    'manufacturer_id'      => DBModels::type('integer'),
    'is_update'            => DBModels::type('boolean'),
    'is_helpdesk_visible'  => DBModels::type('boolean'),
    'is_valid'             => DBModels::type('boolean',
                                             array('visible' => false)),
    'ticket_tco'           => DBModels::type('decimal',
                                             array('visible' => false)),
                                      

);

$table['relationships'] = array(
    'group' => array(
        'type' => 'belongsTo',
        'item' => 'Group'),
    'group_tech' => array(
        'type' => 'belongsTo',
        'item' => 'Group'),
    'location' => array(
        'type' => 'belongsTo',
        'item' => 'Location'),
    'manufacturer' => array(
        'type' => 'belongsTo',
        'item' => 'Manufacturer'),
    'software' => array(
        'type' => 'belongsTo',
        'item' => 'Software'),
    'softwarecategory' => array(
        'type' => 'belongsTo',
        'item' => 'SoftwareCategory'),
    'user' => array(
        'type' => 'belongsTo',
        'item' => 'User'),
    'user_tech' => array(
        'type' => 'belongsTo',
        'item' => 'User'),        
);

include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/software.py
from fusionglpi import db

class Software(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    is_template = db.Column(db.Boolean, nullable=False, server_default="0")
    template_name = db.Column(db.String(255, None, True))
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
    user_tech_id = db.Column(db.Integer, nullable=False, server_default="0")
    group_id = db.Column(db.Integer, db.ForeignKey('group.id'))
    group_tech_id = db.Column(db.Integer, nullable=False, server_default="0")
    location_id = db.Column(db.Integer, db.ForeignKey('location.id'))
    software_id = db.Column(db.Integer, db.ForeignKey('software.id'))
    software_category_id = db.Column(db.Integer, db.ForeignKey('software_category.id'))
    manufacturer_id = db.Column(db.Integer, db.ForeignKey('manufacturer.id'))
    is_update = db.Column(db.Boolean, nullable=False, server_default="0")
    is_helpdesk_visible = db.Column(db.Boolean, nullable=False, server_default="0")
    is_valid = db.Column(db.Boolean, nullable=False, server_default="0")
    ticket_tco = db.Column(db.Float)

    #softwares = db.relationship('Software', backref='software')
    software_licenses = db.relationship('SoftwareLicense', backref='software')
    software_versions = db.relationship('SoftwareVersion', backref='software')